src/api/index.ts

```diff
@@ -187,4 +189,24 @@ export class Api {
       listings,
     };
   }
+
+  async getWrapperAddress(collection: {
+    contractData: { contractAddress: `0x${string}` };
+    wrapperCollections?: { contractData: { contractAddress: `0x${string}` } }[];
+  }) {
+    if (collection.wrapperCollections)
+      return collection.wrapperCollections[0]?.contractData?.contractAddress;
+    const {
+      collection: [
+        {
+          wrapperCollections: [{ contractData: wrapperContractData }],
```

Comment:
Does this work? How? Does it take the first element of the array or what? IMO I’d avoid it as looks super confusing
I’d go the other way around, sth like:

if (!collection)
 const {collection} = await this.api…

const wrapperAddress = collection.wrapperCollections?.[0]….